src/main/java/org/haffson/adventofcode/days/day04/Day04.java

```diff
@@ -0,0 +1,215 @@
+        final List<TimeStampInformation> sortedList = sortListByDate(convertedInput);
+
+        final Map<Integer, List<Integer>> minutesAsleepByGuard = minutesEachGuardIsAsleep(sortedList);
+
+        final Map.Entry<Integer, List<Integer>> guardWithMostMinutesAsleep = findGuardEntryWithMostMinutesAsleep(minutesAsleepByGuard);
+
+        final int minuteWhichGuardIsMostlyAsleep = findMinuteWhichGuardIsMostlyAsleep(guardWithMostMinutesAsleep);
+
+        return guardWithMostMinutesAsleep.getKey() * minuteWhichGuardIsMostlyAsleep;
+    }
+
+    /**
+     * Auxiliary method for Day 4, Part 1.
+     * Converts the String-list to a List of TimeStampInformation.class
+     *
+     * @return List<TimeStampInformation>
+     */
+    List<TimeStampInformation> convertStringListToTimeStampList(final List<String> stringInput) {
+        return stringInput
+                .stream()
+                .map(TimeStampInformation::of)
+                .toList();
+    }
+
+    /**
+     * Auxiliary method for Day 4, Part 1.
+     * Sorts the List<TimeStampInformation> by its LocalDayDate property.
+     *
+     * @return (sorted) List<TimeStampInformation>
+     */
+    List<TimeStampInformation> sortListByDate(final List<TimeStampInformation> stringInput) {
+        return stringInput
+                .stream()
+                .sorted(Comparator.comparing(TimeStampInformation::getTimeStamp))
+                .toList();
+    }
+
+    /**
+     * Auxiliary method for Day 4, Part 1.
+     * Maps the TimeStampInformation's to a Map with Key= GuardID, Value= List of Minutes.
+     * Each List Entry equals a minute.
+     * If a guard is asleep at a certain minute, the value of the minute-index counts up.
+     *
+     * @return Map<Integer, List < Integer>> Map of minutes each guard is asleep
+     */
+    Map<Integer, List<Integer>> minutesEachGuardIsAsleep(final List<TimeStampInformation> sortedList) {
+        int guardId = -1;
+        final Map<Integer, List<Integer>> minutesAsleepByGuard = new HashMap<>();
+        LocalDateTime fallsAsleep = LocalDateTime.MIN;
+
+        for (final TimeStampInformation timeStampInformation : sortedList
+        ) {
+            if (timeStampInformation.getInformation().contains("shift")) {
+                guardId = timeStampInformation.getGuardID();
+            } else if (timeStampInformation.getInformation().contains("falls")) {
+                fallsAsleep = timeStampInformation.getTimeStamp();
+            } else {
+                final LocalDateTime wakesUp = timeStampInformation.getTimeStamp();
+                List<Integer> minutes = minutesAsleepByGuard.get(guardId);
+                if (minutes == null) {
+                    minutes = new ArrayList<>();
+                    for (int i = 0; i < 60; i++) {
+                        minutes.add(i, 0);
+                    }
+                }
+                for (int i = fallsAsleep.getMinute(); i < wakesUp.getMinute(); i++) {
```

Comment:
Hab nicht alles genau gelesen. Reicht es hier nur auf die Minute zu gucken? Was ist wenn zb eine Stunde überschritten wird?

Reply:
In der Aufgabe ist es explizit so geregelt, dass immer nur zwischen 0:00 und 0:59 geschlafen wird